bokeh/client/session.py

```diff
@@ -153,11 +153,36 @@ def __init__(self, session_id=DEFAULT_SESSION_ID, io_loop=None, url=DEFAULT_SERV
         self._connection = ClientConnection(session=self, io_loop=io_loop, url=url)
 
         self._current_patch = None
+        self._callbacks = {}
 
     def _attach_document(self, document):
```

Comment:
Need to load the existing callbacks from the document here in `_attach_document` I think